Dockerfile

```diff
@@ -1,114 +1,122 @@
+    mix local.rebar --force
+
+FROM base as development
 
 # Install mix dependencies
-COPY mix.exs mix.lock $APP_PATH/
+COPY mix.exs mix.lock ./
 RUN mix do deps.get
 
-COPY assets/package.json assets/package-lock.json $APP_PATH/assets/
+# Install npm packages:
+COPY assets/package.json assets/package-lock.json ./assets/
 RUN npm install --prefix assets
 
-COPY . .
+FROM base as production_builder
 
-# Building a release version
-# https://hexdocs.pm/phoenix/releases.html
-FROM elixir_alpine AS production_build
+# set build ENV
+ENV MIX_ENV="prod"
+ENV NODE_ENV="production"
 
-# Set build ENV
-ENV MIX_ENV=prod
+# install mix dependencies
+COPY mix.exs mix.lock ./
+RUN mix deps.get --only $MIX_ENV
+RUN mkdir config
 
-# Install mix dependencies
-COPY mix.exs mix.lock $APP_PATH/
-RUN mix do deps.get, deps.compile
-
-# Build assets
-COPY assets $APP_PATH/assets/
-RUN set -eux ; \
-    npm \
-      --loglevel=error \
-      --no-audit \
-      --prefix assets \
-      --progress=false \
-      ci ; \
-    npm \
-      --prefix assets \
-      run \
-      deploy
-
-# Compile and build release
-COPY . .
-RUN mix do phx.digest, compile, release
-
-# Prepare release image
-FROM alpine:${ALPINE_VERSION} AS production
-
-# Labels Standard
-LABEL org.label-schema.schema-version="1.0"
-# LABEL org.label-schema.build-date=$BUILD_DATE
-LABEL org.label-schema.name="mindwendel/mindwendel"
-LABEL org.label-schema.description="mindwendel application"
-LABEL org.label-schema.vcs-url="https://github.com/mindwendel/mindwendel"
-# LABEL org.label-schema.vcs-ref=$VCS_REF
-LABEL org.label-schema.vendor="WSO2"
-# LABEL org.label-schema.version=$BUILD_VERSION
-LABEL org.label-schema.docker.cmd="docker run -d --name mindwendel -p 127.0.0.1:80:4000 -e DATABASE_HOST=\"...\" -e DATABASE_PORT=\"5432\" -e DATABASE_NAME=\"mindwendel_prod\" -e DATABASE_USER=\"mindwendel_db_user\" -e DATABASE_USER_PASSWORD=\"mindwendel_db_user_password\" -e SECRET_KEY_BASE=\"generate_your_own_secret_key_base_and_save_it\" -e URL_HOST=\"your_domain_to_mindwendel\" ghcr.io/mindwendel/mindwendel"
-LABEL org.opencontainers.image.source="https://github.com/mindwendel/mindwendel"
-LABEL maintainer="gerardo.navarro.suarez@gmail.com"
-
-ENV APP_PATH=/app
-
-RUN apk add --no-cache \
-      libgcc \
-      libstdc++ \
-      ncurses-libs \
-      openssl \
-      postgresql-client
-
-WORKDIR $APP_PATH
-
-RUN chown nobody:nobody $APP_PATH/
-
-USER nobody:nobody
-
-COPY --from=production_build --chown=nobody:nobody $APP_PATH/_build/prod/rel/mindwendel ./
-
-ENV HOME=$APP_PATH
-
-COPY --chown=nobody:nobody entrypoint.release.sh $APP_PATH/entrypoint.release.sh
-ENTRYPOINT ["sh", "entrypoint.release.sh"]
-
-EXPOSE 80
\ No newline at end of file
+# copy compile-time config files before we compile dependencies
+# to ensure any relevant config change will trigger the dependencies
+# to be re-compiled.
+COPY config/config.exs config/${MIX_ENV}.exs config/
+RUN mix deps.compile
+
+COPY priv priv
```

Comment:
I do not expect these many individuals `COPY` commands. I would have expected a final `COPY . .` command that considers the `.dockerignore` file and copies all necesssary files into the docker image. The individual copy command for installing the hex and node packages are understandable.

Is this just a performance improvement to avoid recompiling?